### ISSUE TITLE
Can't read data from OpenTSDB

### ISSUE THREAD

**Issue by johnseekins** [grafana/grafana#13369]
Please include this information:
### What Grafana version are you using?
5.2.4
### What datasource are you using?
OpenTSDB
### What OS are you running grafana on?
Ubuntu 16.04
### What did you do?
Tried to load a graph
### What was the expected result?
A graph displayed
### What happened instead?
Nothing displayed
 
I appear to be hitting the issue described in #4700 (which was closed), and similarly am seeing the behavior that was supposedly fixed in MR #4594.
Basically, using OpenTSDB 2.3.1, I can't read _any_ data in Grafana. It always puts in an empty tag list (not hashmap) that breaks the API.
An example error here:
https://gist.github.com/johnseekins/e74bfc8eeb535a1aa73361389c609e6a

As a side note, your OpenTSDB docs are out of date. They mention selecting `>=2.2` but the options only have `==2.2` and `==2.3`.

**Comment by torkelo** [comment]
Can you open query inspector (top right of metrics tab in Grafana), it shows full query sent and the response from Grafana.

**Comment by johnseekins** [comment]
That was the link I sent in with my request. The full query has `tags: []` in it. Every time.

**Comment by torkelo** [comment]
try deleting the query and create a new one.  I do not get any tags: [] in the queries sent to OpenTSDB

**Comment by johnseekins** [comment]
Every graph I create has the same error in the query inspector. Is it possible that my OpenTSDB plugin didn't update with Grafana?

**Comment by torkelo** [comment]
The only explanation could be that you have a non OpenTSDB datasource as default and that adds a tags property to query object as an array.

The openTSDB query editor will check if there is a tags property and it will add it but it always adds it as an object (if it does not exist):
https://github.com/grafana/grafana/blob/v5.2.4/public/app/plugins/datasource/opentsdb/query_ctrl.ts#L106

**Comment by johnseekins** [comment]
Right. Okay. So deleting and recreating a graph worked this time. I also added OpenTSDB as my "default" data source and the problem _appears_ to persist.

**Comment by torkelo** [comment]
by having as default will make it work for new panels not existing. To fix it for old panels you need to remove add readd query, or export dashboard json and fix query model , then import

**Comment by johnseekins** [comment]
That makes sense. Is there any way to make OpenTSDB _not_ the default and have it work?

**Comment by torkelo** [comment]
Depends on what datasource is default sadly. Seems like you one that adds a tag array to query model

**Comment by johnseekins** [comment]
Okay. Well the important part is it _works_. Thanks for the help!
